[Documentation-Update]: Cross referenced Modules (Bicep) (#1860)
Azure/Azure-Verified-Modules: docs/content/specs-defs/includes/bicep/shared/functional/BCPFR7.md

tags: [
  Class-Resource, # MULTIPLE VALUES: this can be "Class-Resource" AND/OR "Class-Pattern" AND/OR "Class-Utility"
  Type-Functional, # SINGLE VALUE: this can be "Type-Functional" OR "Type-NonFunctional"
  Category-Telemetry, # SINGLE VALUE: this can be "Category-Testing" OR "Category-Telemetry" OR "Category-Contribution/Support" OR "Category-Documentation" OR "Category-CodeStyle" OR "Category-Naming/Composition" OR "Category-Inputs/Outputs" OR "Category-Release/Publishing"
  Language-Bicep, # MULTIPLE VALUES: this can be "Language-Bicep" AND/OR "Language-Terraform"
  Severity-MUST, # SINGLE VALUE: this can be "Severity-MUST" OR "Severity-SHOULD" OR "Severity-MAY"
  Persona-Owner, # MULTIPLE VALUES: this can be "Persona-Owner" AND/OR "Persona-Contributor"
  Persona-Contributor, # MULTIPLE VALUES: this can be "Persona-Owner" AND/OR "Persona-Contributor"
  Lifecycle-BAU, # SINGLE VALUE: this can be "Lifecycle-Initial" OR "Lifecycle-BAU" OR "Lifecycle-EOL"
  Validation-TBD # SINGLE VALUE: this can be "Validation-BCP/Manual" OR "Validation-BCP/CI/Informational" OR "Validation-BCP/CI/Enforced"
]
priority: 10060
---

#### ID: BCPFR7 - Cross-Referencing published Modules

> **Resource modules**, that reference other modules, **MUST** disable the telemetry on the referenced modules.

Reviewer: ```suggestion
**Resource modules**, that reference other modules (child, utility, or other resource modules), **MUST** disable the telemetry on the referenced modules.
```

{{% notice style="note" %}}

> This only applies to resource modules that reference other modules, which are child-modules (if they are published and have a `version.json` file, they have the `enableTelemetry` parameter) or other modules. For pattern modules, [SFR4]({{% siteparam base %}}/spec/SFR4/) applies.

Reviewer: ```suggestion
This only applies to resource modules that reference other modules, such as other resource modules, utility modules or child-modules (if they are published and have a `version.json` file in their directory and they have the `enableTelemetry` parameter). 

For pattern modules, [SFR4]({{% siteparam base %}}/spec/SFR4/) still applies.
```

  Category-Telemetry, # SINGLE VALUE: this can be "Category-Testing" OR "Category-Telemetry" OR "Category-Contribution/Support" OR "Category-Documentation" OR "Category-CodeStyle" OR "Category-Naming/Composition" OR "Category-Inputs/Outputs" OR "Category-Release/Publishing"
  Language-Bicep, # MULTIPLE VALUES: this can be "Language-Bicep" AND/OR "Language-Terraform"
  Severity-MUST, # SINGLE VALUE: this can be "Severity-MUST" OR "Severity-SHOULD" OR "Severity-MAY"
  Persona-Owner, # MULTIPLE VALUES: this can be "Persona-Owner" AND/OR "Persona-Contributor"
  Persona-Contributor, # MULTIPLE VALUES: this can be "Persona-Owner" AND/OR "Persona-Contributor"
  Lifecycle-BAU, # SINGLE VALUE: this can be "Lifecycle-Initial" OR "Lifecycle-BAU" OR "Lifecycle-EOL"
  Validation-TBD # SINGLE VALUE: this can be "Validation-BCP/Manual" OR "Validation-BCP/CI/Informational" OR "Validation-BCP/CI/Enforced"
]
priority: 10060
---

#### ID: BCPFR7 - Cross-Referencing published Modules

**Resource modules**, that reference other modules (child, utility, or other resource modules), **MUST** disable the telemetry on the referenced modules.

{{% notice style="note" %}}

> This only applies to resource modules that reference other modules, such as other resource modules, utility modules or child-modules (if they are published and have a `version.json` file in their directory and they have the `enableTelemetry` parameter).

Reviewer 2: I'd suggest to elaborate by splitting the sentence
```suggestion
This only applies to resource modules that reference other modules, such as:
- other resource modules
- utility modules
- child-modules qualifying for publishing, i.e. having a `version.json` file in their directory and exposing the `enableTelemetry` input parameter.
```

> This only applies to resource modules that reference other modules, such as other resource modules, utility modules or child-modules (if they are published and have a `version.json` file in their directory and they have the `enableTelemetry` parameter). 

Reviewer: ```suggestion
This only applies to resource modules that reference other modules, such as other resource modules, utility modules or child-modules (if they are published and have a `version.json` file in their directory and they have the `enableTelemetry` parameter).
```

For pattern modules, [SFR4]({{% siteparam base %}}/spec/SFR4/) still applies.

{{% /notice %}}

> A variable named `enableReferencedModulesTelemetry` is created in the main.bicep file of the module, that cross-refernces other published modules, and set to `false`. This variable is used to set the `enableTelemetry` parameter of cross-referenced modules.

Reviewer 2: ```suggestion
A variable named `enableReferencedModulesTelemetry` is created in the main.bicep file of the module, that cross-references other published modules, and set to `false`. This variable is used to set the `enableTelemetry` parameter of cross-referenced modules.
```
